Framework/DataHandling/src/LoadILLReflectometry.cpp

```diff
@@ -62,406 +197,617 @@ int LoadILLReflectometry::confidence(
-
-  std::string instrumentPath = m_loader.findInstrumentNexusPath(firstEntry);
-  setInstrumentName(firstEntry, instrumentPath);
-
-  initWorkSpace(firstEntry, monitorsData);
-
-  g_log.debug("Building properties...");
-  loadNexusEntriesIntoProperties(filenameData);
-
-  g_log.debug("Loading data...");
-  loadDataIntoTheWorkSpace(firstEntry, monitorsData);
-
-  // load the instrument from the IDF if it exists
-  g_log.debug("Loading instrument definition...");
-  runLoadInstrument();
-
-  // 1) Move
-
-  // Get distance and tilt angle stored in nexus file
-  // Mantid way
-  ////	auto angleProp =
-  /// dynamic_cast<PropertyWithValue<double>*>(m_localWorkspace->run().getProperty("dan.value"));
-  // Nexus way
-  double angle =
-      firstEntry.getFloat("instrument/dan/value"); // detector angle in degrees
-  double distance = firstEntry.getFloat(
-      "instrument/det/value"); // detector distance in millimeter
-
-  distance /= 1000.0; // convert to meter
-  g_log.debug() << "Moving detector at angle " << angle << " and distance "
-                << distance << '\n';
-  placeDetector(distance, angle);
-
-  // Set the channel width property
-  auto channel_width = dynamic_cast<PropertyWithValue<double> *>(
-      m_localWorkspace->run().getProperty("monitor1.time_of_flight_0"));
-  m_localWorkspace->mutableRun().addProperty<double>(
-      "channel_width", *channel_width, true); // overwrite
-
-  // Set the output workspace property
-  setProperty("OutputWorkspace", m_localWorkspace);
-}
-
 /**
- * Set member variable with the instrument name
- */
-void LoadILLReflectometry::setInstrumentName(
-    const NeXus::NXEntry &firstEntry, const std::string &instrumentNamePath) {
+  * Init names of member variables based on instrument specific NeXus file
+  * entries
+  *
+  * @param entry :: the NeXus file entry
+  */
+void LoadILLReflectometry::initNames(NeXus::NXEntry &entry) {
+  std::string instrumentNamePath = m_loader.findInstrumentNexusPath(entry);
+  m_instrumentName = entry.getString(instrumentNamePath.append("/name"));
+  if (m_instrumentName.empty())
+    throw std::runtime_error(
+        "Cannot set the instrument name from the Nexus file!");
+  // In NeXus files names are: D17 and figaro. The instrument
+  // definition is independent and names start with a capital letter. This
+  // loader follows its convention.
+  boost::to_lower(m_instrumentName);
+  m_instrumentName[0] = char((std::toupper(m_instrumentName[0])));
+  g_log.debug() << "Instrument name: " << m_instrumentName << '\n';
+  if (m_instrumentName == "D17") {
+    m_detectorDistance = "det";
+    m_detectorAngleName = "dan.value";
+    m_sampleAngleName = "san.value";
+    m_offsetFrom = "VirtualChopper";
+    m_offsetName = "open_offset";
+    m_pixelCentre = 127.5;
+    m_chopper1Name = "Chopper1";
+    m_chopper2Name = "Chopper2";
+  } else if (m_instrumentName == "Figaro") {
+    // TODO Figaro's detector position should be calculated from
```

Comment:
If these TODOs are done, could they be removed?

Reply:
This loader loads current Figaro files incorrectly, and the TODOs are there to remind us about it. In fact, the file formats for both reflectometers at ILL will change in the near future, and we already have plans for a large overhaul of the loader, and an update to version 2. Meanwhile, however, we need something that can load D17 files. This enables us to build the data reduction workflow for the reflectometers.

Reply:
Yup, I see. In that case, In that case everything looks good!